sharding/notary/notary.go

```diff
@@ -151,7 +205,287 @@ func joinNotaryPool(client mainchain.Client) error {
 	if err != nil {
 		return fmt.Errorf("unable to deposit eth and become a notary: %v", err)
 	}
+	receipt, err := client.TransactionReceipt(tx.Hash())
+	if err != nil {
+		return err
+	}
+	if receipt.Status == uint(0) {
+
+		return errors.New("Transaction was not successful, unable to deposit ETH and become a notary")
+
+	}
+
+	if inPool, err := isAccountInNotaryPool(client); !inPool || err != nil {
+		if !inPool {
```

Comment:
This is weird. If `isAccountInNotaryPool` has an error, then inPool will be false, but you are not handing the error.

Please refactor this to handle the error first then handle the bool value.